### PR TITLE
feat: add standard endpoint to retrieve fork choice context

### LINKED ISSUE #5700: Implement debug fork choice API
### Problem description

Latest beacon API spec requires debug fork choice API
- https://github.com/ethereum/beacon-APIs/pull/232

### Solution description

Implement the following route:
- https://ethereum.github.io/beacon-APIs/#/Debug/getDebugForkChoice

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nflaig**
**Motivation**

- https://github.com/ethereum/beacon-APIs/pull/232

**Description**

Add standardized endpoint from beacon-api spec to retrieve fork choice context

Closes  https://github.com/ChainSafe/lodestar/issues/5700

### CODE REVIEW

File: packages/api/src/beacon/routes/debug.ts [hunk 1]
```diff
@@ -115,6 +150,24 @@ export function getDefinitions(_config: ChainForkConfig): RouteDefinitions<Endpo
...
     },
+    getDebugForkChoice: {
+      url: "/eth/v1/debug/fork_choice",
+      method: "GET",
+      req: EmptyRequestCodec,
+      resp: {
+        data: ForkChoiceResponseType,
+        meta: EmptyMetaCodec,
+        onlySupport: WireFormat.json,
+        transform: {
```
Comment (nflaig) [inline packages/api/src/beacon/routes/debug.ts:161]: as per spec it's not wrapped in `data` container... it's too late to change it now though :cry:

File: packages/beacon-node/src/api/rest/base.ts [hunk 2]
```diff
@@ -96,7 +96,7 @@ export class RestApiServer {
       if (err.validation) {
         const {instancePath, message} = err.validation[0];
         const payload: ErrorResponse = {
-          code: err.statusCode ?? 400,
```
Comment (wemeetagain) [inline packages/beacon-node/src/api/rest/base.ts:99]: was this intended?

Comment (nflaig) [reply packages/beacon-node/src/api/rest/base.ts:99]: It should probably not be part of this PR, but noticed using statusCode of error does not make sense here as it's always gonna be 400 (also to match line 103)

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.23.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.23.0) :tada:

**Pull request merged**